azurecompute-arm/src/test/java/org/jclouds/azurecompute.arm/features/ResourceGroupApiLiveTest.java

```diff
@@ -0,0 +1,131 @@
+import static org.testng.Assert.assertEquals;
+
+
+@Test(groups = "live", testName = "ResourceGroupApiLiveTest")
+public class ResourceGroupApiLiveTest extends BaseAzureComputeApiLiveTest {
+   private String resourcegroup;
+
+   @BeforeClass
+   @Override
+   public void setup(){
+      super.setup();
+      resourcegroup = getResourceGroupName();
+   }
+
+   private ResourceGroupApi api() {
+      return api.getResourceGroupApi();
+   }
+
+   @Test(dependsOnMethods = "testCreate")
+   public void testList() {
+      final List<ResourceGroup> resourceGroups = api().list();
+
+      assertTrue(resourceGroups.size() > 0);
+
+      assertTrue(Iterables.any(resourceGroups, new Predicate<ResourceGroup>() {
+
+         @Override
+         public boolean apply(final ResourceGroup group) {
+            return resourcegroup.equals(group.name());
+         }
+      }));
+   }
+
+   @Test(dependsOnMethods = "testCreate")
+   public void testRead() {
+      final ResourceGroup group = api().get(resourcegroup);
+      assertNotNull(group);
+      assertEquals(group.name(), resourcegroup);
+      assertEquals(group.location(), LOCATION);
+   }
+
+   public void testCreate() {
+
+      final ResourceGroup resourceGroup = api().create("jcloudstest", LOCATION, null);
+      assertEquals(resourceGroup.name(), "jcloudstest");
+      assertEquals(resourceGroup.location(), LOCATION);
+      assertEquals(resourceGroup.tags().size(), 0);
+      assertTrue(resourceGroup.id().contains("jcloudstest"));
+      assertEquals(resourceGroup.properties().provisioningState(), "Succeeded");
+   }
+
+   @Test(dependsOnMethods = "testCreate")
+   public void testUpdateWithEmptyTag() {
+      ImmutableMap<String, String> tags = ImmutableMap.<String, String>builder().build();
+
+      final ResourceGroup resourceGroup = api().update("jcloudstest", tags);
+
+      assertEquals(resourceGroup.tags().size(), 0);
+      assertEquals(resourceGroup.properties().provisioningState(), "Succeeded");
+   }
+
+   @Test(dependsOnMethods = "testCreate")
+   public void testUpdateWithTag() {
+      ImmutableMap<String, String> tags = ImmutableMap.<String, String>builder().put("test1", "value1").build();
+
+      final ResourceGroup resourceGroup = api().update("jcloudstest", tags);
+
+      assertEquals(resourceGroup.tags().size(), 1);
+      assertEquals(resourceGroup.properties().provisioningState(), "Succeeded");
+   }
+
+   @AfterClass(alwaysRun = true)
+   public void testDelete() throws Exception {
+      URI uri =  api().delete(resourcegroup);
+      assertNotNull(uri);
+      assertTrue(uri.toString().contains("api-version"));
+      assertTrue(uri.toString().contains("operationresults"));
+
+      JobStatus status = api.getJobApi().jobStatus(uri);
+      assertEquals(status, JobStatus.IN_PROGRESS);
+      while (status == JobStatus.IN_PROGRESS){
+         status = api.getJobApi().jobStatus(uri);
+         if (status == JobStatus.DONE){
+            break;
+         }
+      }
```

Comment:
Do delete operations take a considerable amount of time? Would it make sense to timeout the operation at some point? If that makes sense, a common and straightforward pattern to do that in jclouds is to use one of the [Predicates2.retry](http://jclouds.apache.org/reference/javadoc/1.9.x/org/jclouds/util/Predicates2.html) variants. Something like:
```java
boolean jobDone = Predicates2.retry(new Predicate<URI>() {
   @Override public boolean apply(URI uri) {
      return JobStatus.DONE == api.getJobApi().jobStatus(uri);
   }
}, 60 * 2 * 1000 /* 2 minutes timeout */).apply(uri);
assertTrue(jobDone, "delete operation did not complete in the configured timeout");
```